src/slicer.h

```diff
@@ -109,12 +116,350 @@ class SlicerLayer
 
     /*!
      * Try to close up polylines into polygons while they have large gaps in them.
-     * 
+     *
      * Clears all open polylines which are used up in the process
-     * 
+     *
      * \param[in,out] open_polylines The polylines which are stiched, but couldn't be closed into a loop yet
      */
     void stitch_extensive(Polygons& open_polylines);
+
+private:
+    /*!
+     * \brief This class represents the location of an end point of a
+     * polyline in a polyline vector.
+     *
+     * The location records the index in the polyline vector and
+     * whether this is the vertex at the start of the polyline or the
+     * vertex at the end.
+     */
+    class Terminus
+    {
+    public:
+        /*! A representation of Terminus that can be used as an array index. */
```

Comment:
The Terminus is encoding 2 things.  The first is the index into open_polylines.  This is what is returned by getPolylineIdx().  The second is whether it is referring to the start (polyline[0]) or end (polyline.back()) of the polyline.

No, that's not quite correct.  asIndex() is used to index the std::vector's which implement the maps between old and current terminus.  The Index type is non-negative, contiguous, and has a well defined past the end index.  It doesn't strictly need to be contiguous, but it needs to be mostly so to be efficient.  It doesn't matter that the Index type keeps the same ordering as the polyline indices.